### PR TITLE
[Privatization] Skip with encapsed string part on ChangeReadOnlyVariableWithDefaultValueToConstantRector

### PR DESCRIPTION
**Pull request by samsonasik**
Closes #1282  Fixes https://github.com/rectorphp/rector/issues/6818

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_with_encapsed_string_part.php.inc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Rector\Tests\Privatization\Rector\Class_\ChangeReadOnlyVariableWithDefaultValueToConstantRector\Fixture;
+
+final class SkipWithEncapsedStringPart
+{
+    public function run(int $id): array
+    {
+        $data                = [
+            'url' => "/$id.pdf",
+        ];
```
Comment (TomasVotruba) [inline rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_with_encapsed_string_part.php.inc:11]: What about use case with $date = "$id";?

Comment (samsonasik) [reply rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_with_encapsed_string_part.php.inc:11]: implemented https://github.com/rectorphp/rector-src/pull/1287/commits/97ecef4a864eef28c9940cc4141774d2a0f1cd6a

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Comment by cavo789** [comment]
Thanks @samsonasik

**Pull request merged**